aiogram/filters/callback_data.py

```diff
@@ -120,9 +120,8 @@ def unpack(cls: Type[T], value: str) -> T:
             raise ValueError(f"Bad prefix ({prefix!r} != {cls.__prefix__!r})")
         payload = {}
         for k, v in zip(names, parts):  # type: str, Optional[str]
-            if field := cls.model_fields.get(k):
-                if v == "" and not field.is_required():
-                    v = None
+            if v == "":
```

Comment:
This solution breaks case when field is required with any type (`demo: str` typehint) and value is empty (`demo == ""`)


You should check that typehint of the field is Union and has None argument instead of checking `field.is_required()`

```python
typing.get_origin(field.annotation) is types.UnionType 
and type(None) in typing.get_args(field.annotation)
```

But this solution don't wor in Python <=3.9 because `types.UnionType` is introduced [in the 3.10 version](https://docs.python.org/3/library/types.html#types.UnionType).

We should research alternative universal solution for all currently supported versions of the Python.